Makefile

```diff
@@ -0,0 +1,39 @@
+RESET_TARGET := upstream/master
+override PKGS := $(shell go list ./... | grep -v /vendor/)
+override BRANCH := $(shell sh -c "git branch | sed -n 's|* \(.*\)|\1|p'")
+override STAGED_GO_FILES := $(shell git diff --name-only --cached | grep -E ".*.go")
+
+.PHONY: squash format fmt dep test docker/test
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+ifeq ($(COMMIT),)
+squash:
+	@echo "COMMIT=<COMMIT> not provided. Exiting..."
+	@exit 1
+else
+
+# Assuming PR is ready, everything added as intended, no conflicts
+squash: docker/test format ## Runs dockerized test, formats, and force pushes. Override RESET_TARGET='upstream/master' if necessary
```

Comment:
This is pretty scary, I'd honestly not use this simply because it's involving a force push. I'd much rather make a copy of the user's current branch and force push the changes to that. A user can then, if that worked, force push from their copied branch to the main one. Then there's no chance of disaster

Reply:
Removing pushing logic. Commit squashing only on test success